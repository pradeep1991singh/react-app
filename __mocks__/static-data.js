// @flow

import type { Store } from '../src/store/type-store'

export const userStore = {
  isLoading: false,
  user: {
    authToken: 'authToken',
    userDetails: {
      apiKey: 'apiKey',
      email: 'email',
      firstName: 'firstName',
      lastName: 'lastName',
      mobileNumber: 1234567890,
      userId: 'userId'
    },
    onlineUsers: {
      userId: 'userId'
    },
    currentReceiver: {
      receiverId: 'receiverId',
      receiverName: 'receiverName'
    },
    error: {
      code: '',
      message: ''
    }
  }
}

export function getStore(store?: Store) {
  return {
    getState() {
      return {
        userStore: userStore,
        dispatch() {
          // $FlowFixMe
          return jest.fn()
        },
        subscribe() {
          // $FlowFixMe
          return jest.fn()
        },
      }
    },
  }
}
