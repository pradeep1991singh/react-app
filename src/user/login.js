// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { toast } from 'react-toastify'

import { loginAction } from './user-store'
import { LOGGING_IN } from '../common'
import './user.css'

import type { LoginProps, LoginStates, User } from './type-user'

export class Login extends Component<LoginProps, LoginStates> {

    state = {
        email: '',
        password: ''
    }

    componentWillMount() {
        this.validateUser(this.props.user, this.props.errorMessage)
    }

    componentWillReceiveProps(nextProps: LoginProps) {
        if (nextProps !== this.props) {
            const { user, errorMessage } = nextProps
            this.validateUser(user, errorMessage)
        }
    }

    validateUser = (user: User, errorMessage: string) => {
        if (Object.keys(user).length > 0) {
            toast.success(LOGGING_IN, {
                position: toast.POSITION.TOP_CENTER,
            })
            this.props.history.push({ pathname: '/chat' })
        }

        if (errorMessage) {
            toast.error(errorMessage, {
                position: toast.POSITION.TOP_CENTER,
            })
        }
    }

    handleInputChange = (event: any) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;

        this.setState({
            [name]: value
        });
    }

    handleSubmit = (event: any) => {
        event.preventDefault();
        this.props.loginAction(this.state)
    }

    render() {
        const { email, password } = this.state
        return (
            <div className="form-container">
                <h4 className="page-title">Login</h4>
                <form name="login" onSubmit={this.handleSubmit}>
                    <input type="email"
                        name="email"
                        value={email}
                        required
                        placeholder="Email"
                        onChange={this.handleInputChange} />

                    <input type="password"
                        name="password"
                        value={password}
                        required
                        placeholder="Password"
                        onChange={this.handleInputChange} />

                    <input type="submit" value="Login" />
                </form>
            </div>
        )
    }
}

const mapStateToProps = ({ userStore: { user, error: { message: errorMessage } } }) => ({
    user,
    errorMessage
})

const mapDispatchToProps = dispatch =>
    bindActionCreators(
        {
            loginAction
        },
        dispatch
    )

export default connect(mapStateToProps, mapDispatchToProps)(Login)
