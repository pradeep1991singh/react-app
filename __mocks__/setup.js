
import fetch from './fetch-mock';
import { userStore } from './static-data'

// mock fetch
global.fetch = fetch;

// mock localStorage
const localStorageMock = (function () {
    const store = {};
    return {
        getItem: function (key) {
            return store[key] || JSON.stringify(userStore.user)
        },
        setItem: function (key, value) {
            store[key] = value.toString()
        },
        removeItem: function () {
            delete store[key]
        }
    };
})();

global.localStorage = localStorageMock
