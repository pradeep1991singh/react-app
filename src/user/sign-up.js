// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { toast } from 'react-toastify'

import { signUpAction } from './user-store'
import { LOGGING_IN } from '../common'
import './user.css'

import type { SignUpProps, SignUpStates } from './type-user'

export class SignUp extends Component<SignUpProps, SignUpStates> {

    state = {
        firstName: '',
        lastName: '',
        email: '',
        mobileNumber: '',
        password: '',
        confirmPassword: ''
    }

    componentWillReceiveProps(nextProps: SignUpProps) {
        if (nextProps !== this.props) {
            const { user, errorMessage } = nextProps
            if (Object.keys(user).length > 0) {
                toast.success(LOGGING_IN, {
                    position: toast.POSITION.TOP_CENTER,
                })
                this.props.history.push({
                    pathname: '/chat',
                    params: user
                })
            }

            if (errorMessage) {
                toast.error(errorMessage, {
                    position: toast.POSITION.TOP_CENTER,
                })
            }
        }
    }


    handleInputChange = (event: any) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;

        this.setState({
            [name]: value
        });
    }

    handleSubmit = (event: any) => {
        event.preventDefault();
        const { password, confirmPassword } = this.state
        if (password !== confirmPassword) {
            toast.error(`Password doesn't match.`, {
                position: toast.POSITION.TOP_CENTER,
            })
        } else {
            this.props.signUpAction(this.state)
        }
    }

    render() {
        const { firstName, lastName, email, mobileNumber, password, confirmPassword } = this.state

        return (
            <div className="form-container">
                <h4>Sign Up</h4>
                <form name="sign-up" onSubmit={this.handleSubmit}>
                    <input type="text"
                        name="firstName"
                        value={firstName}
                        required
                        placeholder="First Name"
                        onChange={this.handleInputChange} />

                    <input type="text"
                        name="lastName"
                        value={lastName}
                        required
                        placeholder="Last Name"
                        onChange={this.handleInputChange} />

                    <input type="email"
                        name="email"
                        value={email}
                        required
                        placeholder="Email"
                        onChange={this.handleInputChange} />

                    <input type="tel"
                        name="mobileNumber"
                        value={mobileNumber}
                        required
                        placeholder="Mobile Number"
                        onChange={this.handleInputChange} />

                    <input type="password"
                        name="password"
                        value={password}
                        required
                        placeholder="Password"
                        onChange={this.handleInputChange} />

                    <input type="password"
                        name="confirmPassword"
                        value={confirmPassword}
                        required
                        placeholder="Confirm Password"
                        onChange={this.handleInputChange} />

                    <input type="submit" value="Sign Up" />
                </form>
            </div >
        )
    }
}

const mapStateToProps = ({ userStore: { user, error: { message: errorMessage } } }) => ({
    user,
    errorMessage
})

const mapDispatchToProps = dispatch =>
    bindActionCreators(
        {
            signUpAction
        },
        dispatch
    )

export default connect(mapStateToProps, mapDispatchToProps)(SignUp)
