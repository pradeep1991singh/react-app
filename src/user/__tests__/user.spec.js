// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import { Login } from '../login'
import { SignUp } from '../sign-up'
import { getStore, userStore } from '../../../__mocks__/static-data'

describe('User', () => {
    const store = getStore()

    const props = {
        user: userStore.user,
        errorMessage: '',
        history: {
            push: jest.fn(),
        },
        loginAction: jest.fn()
    }

    it('should call render Login Component properly', () => {
        const login = renderer.create(
            <Login {...props} />
        ).toJSON()
        expect(login).toMatchSnapshot()
    })

    it('should call render SignUp Component properly', () => {
        const signUp = renderer.create(
            <SignUp user={userStore.user} errorMessage={''} />
        ).toJSON()
        expect(signUp).toMatchSnapshot()
    })
})
