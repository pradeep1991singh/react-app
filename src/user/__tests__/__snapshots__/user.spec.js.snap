// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should call render Login Component properly 1`] = `
<div
  className="form-container"
>
  <h4
    className="page-title"
  >
    Login
  </h4>
  <form
    name="login"
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;

exports[`User should call render SignUp Component properly 1`] = `
<div
  className="form-container"
>
  <h4>
    Sign Up
  </h4>
  <form
    name="sign-up"
    onSubmit={[Function]}
  >
    <input
      name="firstName"
      onChange={[Function]}
      placeholder="First Name"
      required={true}
      type="text"
      value=""
    />
    <input
      name="lastName"
      onChange={[Function]}
      placeholder="Last Name"
      required={true}
      type="text"
      value=""
    />
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      name="mobileNumber"
      onChange={[Function]}
      placeholder="Mobile Number"
      required={true}
      type="tel"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <input
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Confirm Password"
      required={true}
      type="password"
      value=""
    />
    <input
      type="submit"
      value="Sign Up"
    />
  </form>
</div>
`;
